atom/browser/lib/guest-view-manager.coffee

```diff
@@ -38,12 +39,23 @@ moveLastToFirst = (list) ->
 getNextInstanceId = (webContents) ->
   ++nextInstanceId
 
+# Generate URL encoded partition id.
+getPartitionId = (partition) ->
+  persist = partition.startsWith('persist:')
+  # Guest site url will be chrome-guest://fake-host/{persist}?{partitionId}
+  partitionId = "chrome-guest://fake-host/"
+  if partition
+    partitionId += if persist then 'persist?' else '?'
+    partitionId += crypto.createHash('sha256').update(partition).digest('hex')
```

Comment:
There is [a bug](https://github.com/atom/electron/issues/2073) that crypto is slow on first run, maybe we should just use partition's name in lower case with punctuations removed? This way it is easy to find out what's under the data dir, we would also have to notice users about valid partition names though.